test/src/test/java/lib/form/PasswordTest.java

```diff
@@ -62,25 +57,43 @@
 import org.kohsuke.stapler.Stapler;
 import org.kohsuke.stapler.StaplerRequest;
 
-public class PasswordTest extends HudsonTestCase implements Describable<PasswordTest> {
-    public Secret secret;
+public class PasswordTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
 
-    public void test1() throws Exception {
-        secret = Secret.fromString("secret");
-        HtmlPage p = createWebClient().goTo("self/test1");
```

Comment:
Oh thank you, these `self` references from `HudsonTestCase`s were a pesky impediment to switching to `JenkinsRule`.